### PR TITLE
post-startup.sh code cleanup

### PR DESCRIPTION
**Pull request by mbookman**
* Sets several file path variables used through
* Emits messages to the log prior to major operations
* Removes a directory change mid-script (in SSH setup)
* Minor logical reordering of a few steps

Biggest changes are to SSH setup
* Uses best practices around directory and file permissions
* Sets the private key without the key showing up in the post-startup log file
* Removes an "ssh-add" that has always been quietly failing

### CODE REVIEW

File: service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh [hunk 1]
```diff
@@ -305,29 +331,37 @@ EOF
...
-if [[ -n "$TERRA_SSH_KEY" ]]; then
-  printf '%s' "$TERRA_SSH_KEY" | sudo -u "${JUPYTER_USER}" sh -c "jq -r '.privateSshKey' > .ssh/id_rsa"
-  sudo -u "${JUPYTER_USER}" sh -c 'chmod go-rwx .ssh/id_rsa'
-  sudo -u "${JUPYTER_USER}" sh -c 'ssh-add .ssh/id_rsa; ssh-keyscan -H github.com >> ~/.ssh/known_hosts'
+# Create the user SSH directory 
+sudo -u "${JUPYTER_USER}" sh -c "mkdir -p ${USER_SSH_DIR} --mode 0700"
+
+# Get the user's SSH key from Terra, and if set, write it to the user's .ssh directory
+sudo -u "${JUPYTER_USER}" sh -c "\
+  install --mode 0600 /dev/null ${USER_SSH_DIR}/id_rsa.tmp && \
```
Comment (wnojopra) [inline service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh:349]: Could you please explain why we need to do this tmp file work?

**Review by wnojopra — APPROVED** [review]
Great changes! LGTM with one question

**Review by rogerwangcs — APPROVED** [review]
LGTM, assuming thread on testing new user without ssh key is resolved.

**Comment by mbookman** [comment]
Just pushed a change with a small fix, after testing the case where I've deleted my SSH keys from my Terra profile.

**Pull request merged**